content/en/blog/2022/knative.md

```diff
@@ -0,0 +1,123 @@
+
+Follows log output with request headers from the first service:
+```bash
+2022/02/17 12:53:48 Request headers:
+2022/02/17 12:53:48 	X-B3-Sampled: [1]
+2022/02/17 12:53:48 	X-B3-Spanid: [af6c239eb7b39349]
+2022/02/17 12:53:48 	X-B3-Traceid: [5f2c4775e0e36efc1d554a0b6c456cc1]
+2022/02/17 12:53:48 	X-Forwarded-For: [10.244.0.12, 10.244.0.5]
+2022/02/17 12:53:48 	Accept-Language: [en,fr;q=0.9,de;q=0.8,sk;q=0.7]
+2022/02/17 12:53:48 	Cookie: [_ga=GA1.2.260863911.1644918876]
+2022/02/17 12:53:48 	Accept: [text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9]
+2022/02/17 12:53:48 	K-Proxy-Request: [activator]
+2022/02/17 12:53:48 	Upgrade-Insecure-Requests: [1]
+2022/02/17 12:53:48 	User-Agent: [Mozilla/5.0 (X11; Fedora; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.80 Safari/537.36]
+2022/02/17 12:53:48 	X-Request-Id: [ee2797b5-1ee9-408e-b1ff-d5e5431977e6]
+2022/02/17 12:53:48 	Cache-Control: [max-age=0]
+2022/02/17 12:53:48 	X-Forwarded-Proto: [http]
+2022/02/17 12:53:48 	Traceparent: [00-5f2c4775e0e36efc1d554a0b6c456cc1-af6c239eb7b39349-01]
+2022/02/17 12:53:48 	Accept-Encoding: [gzip, deflate]
+2022/02/17 12:53:48 	Forwarded: [for=10.244.0.12;proto=http]
+2022/02/17 12:53:48 Response headers:
+2022/02/17 12:53:48 	Traceparent: [00-5f2c4775e0e36efc1d554a0b6c456cc1-1cf3f827eba96bf2-01]
+2022/02/17 12:53:48
+```
+
+Now let's take a look at logging from the second service which exposes API to consume Knative events. The event API in this case is just an HTTP endpoint which is a cloud event implementation detail:
+```bash
+2022/02/17 13:39:36 Event received: Context Attributes,
+  specversion: 1.0
+  type: httpbody
+  source: github/com/pavolloffay
+  id: fad4139c-b3fb-48b2-b0f4-fee44addc5f1
+  time: 2022-02-17T13:39:34.426355726Z
+  datacontenttype: text/plain
+Extensions,
+  knativearrivaltime: 2022-02-17T13:39:34.491325425Z
+Data,
+  hello from first, traceid=84da42ff2a26bda453330b23dde1a898
+```
+
+We see that the trace-context is not directly present in the event object. However, it is encoded in the incoming transport message - HTTP headers.
+
+### Future improvements
+
+In the previous section, it was mentioned that the Knative serving and eventing components are instrumented with OpenCensus SDK. The instrumentation will change in the future to OpenTelemetry which is tracked in [knative/eventing/issues/3126](https://github.com/knative/eventing/issues/3126) and [knative/pkg/issues/855](https://github.com/knative/pkg/issues/855). The SDK change might not have an immediate impact on the user, however, it will enable users to start natively reporting data in OpenTelemetry format (OTLP).
```

Comment:
```suggestion
In the previous section, it was mentioned that the Knative serving and eventing components are instrumented with OpenCensus SDK. The instrumentation will change in the future to OpenTelemetry which is tracked in [knative/eventing#3126](https://github.com/knative/eventing/issues/3126) and [knative/pkg#855](https://github.com/knative/pkg/issues/855). The SDK change might not have an immediate impact on the user, however, it will enable users to start natively reporting data in OpenTelemetry format (OTLP).
```

nit: github formatting of issues links